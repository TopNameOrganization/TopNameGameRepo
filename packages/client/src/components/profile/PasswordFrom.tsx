import * as React from 'react';
import {useFormik} from 'formik';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import {validationUserPassword} from "./validation";
import {UserPasswordData} from "../../api/types";
import UsersAPI from "../../api/UsersAPI";

export const PasswordForm = (props: any) => {

  const formik = useFormik({
    initialValues: {
      oldPassword: '',
      newPassword: ''
    },
    validationSchema: validationUserPassword,
    onSubmit: (passwordData: UserPasswordData) => {
      UsersAPI.changeUserPassword(passwordData)
      props.noUseForm()
    },
  });

  return (
      <Box id="passwordForm" component="form" onSubmit={formik.handleSubmit} noValidate sx={{ mt: 1 }}>
        <TextField
            variant="standard"
            margin="dense"
            required
            fullWidth
            name="oldPassword"
            label="Old password"
            type="password"
            id="oldPassword"
            autoComplete="oldPassword"
            value={formik.values.oldPassword}
            onChange={formik.handleChange}
            error={formik.touched.oldPassword && Boolean(formik.errors.oldPassword)}
            helperText={formik.touched.oldPassword && formik.errors.oldPassword}
        />
        <TextField
            variant="standard"
            margin="dense"
            required
            fullWidth
            name="newPassword"
            label="New password"
            type="password"
            id="newPassword"
            autoComplete="newPassword"
            value={formik.values.newPassword}
            onChange={formik.handleChange}
            error={formik.touched.newPassword && Boolean(formik.errors.newPassword)}
            helperText={formik.touched.newPassword && formik.errors.newPassword}
        />
        <Button
            type="submit"
            fullWidth
            variant="contained"
            sx={{ mt: 5, mb: 2 }}
        >
          Save Password
        </Button>
      </Box>
  );
}
