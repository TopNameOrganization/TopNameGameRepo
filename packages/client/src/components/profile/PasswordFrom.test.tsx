import * as renderer from 'react-test-renderer';
import { PasswordForm } from './PasswordFrom';

describe('Password form', () => {
  const useForm = true;

  it('should renders password form correctly', () => {
    const tree = renderer
        .create(<PasswordForm noUseForm={useForm} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
