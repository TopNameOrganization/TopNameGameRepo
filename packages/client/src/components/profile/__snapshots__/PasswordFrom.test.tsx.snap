// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password form should renders password form correctly 1`] = `
<form
  className="MuiBox-root css-164r41r"
  id="passwordForm"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="oldPassword"
      id="oldPassword-label"
    >
      Old password
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="oldPassword"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="oldPassword"
        name="oldPassword"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="newPassword"
      id="newPassword-label"
    >
      New password
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="newPassword"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="newPassword"
        name="newPassword"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1p8cus0-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    Save Password
  </button>
</form>
`;
